compiler/passes/resolveIntents.cpp

```diff
@@ -253,12 +254,62 @@ IntentTag blankIntentForExternFnArg(Type* type) {
     return INTENT_CONST_IN;
 }
 
+// Generate an unstable warning if a `const` argument would be transformed into
+// `const ref` and the argument is implicitly modified over the course of the
+// function.  We may decide to change this behavior in the future.
+static void warnForConstIntent(ArgSymbol* arg) {
+  SET_LINENO(arg);
+
+  FnSymbol* fn = toFnSymbol(arg->defPoint->parentSymbol);
+  if (fn->body->length() == 0) {
+    // Exit early if the body is empty - this can happen when
+    // concreteIntentForArg is called during function instantiation, as the
+    // function body won't have been populated yet.  Such functions will have
+    // resolveIntents called on them later
+    return;
+  }
+
+  // Hash the argument at the start of the function
+  CallExpr* getStartHash = new CallExpr(PRIM_CONST_ARG_HASH, new SymExpr(arg));
+
+  VarSymbol* startHash = newTemp(dtUInt[INT_SIZE_DEFAULT]);
```

Comment:
IMO you should use `dtUInt[INT_SIZE_64]` here since we're working with C code that specifies 64 bits.